spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/ActuatorOperations.java

```diff
@@ -0,0 +1,89 @@
+ *
+ * @author David Turanski
+ */
+public interface ActuatorOperations {
+
+	/**
+	 * Get a resource from an actuator path.
+	 * @param deploymentId the deployment ID of the deployed app.
+	 * @param instanceIndex index of the app instance.
+	 * @param endpoint the endpoint path relative to the base actuator URL for the instance.
+	 * @param responseType the expected response type.
+	 * @return the contents as the given type.
+	 */
+	<T> T getFromActuator(String deploymentId, int instanceIndex, String endpoint, Class<T> responseType);
+
+	/**
+	 * Get a resource from an actuator path, for single app instance with index 0.
+	 * @param deploymentId the deployment ID of the deployed app.
+	 * @param endpoint the endpoint path relative to the base actuator URL for the instance.
+	 * @param responseType the expected response type.
+	 * @return the contents as the given type.
+	 */
+	default <T> T getFromActuator(String deploymentId, String endpoint, Class<T> responseType) {
+		return getFromActuator(deploymentId, 0, endpoint, responseType);
+	}
+
+	/**
+	 * Get a resource from an actuator path.
+	 * @param deploymentId the deployment ID of the deployed app.
+	 * @param instanceIndex index of the app instance.
+	 * @param endpoint the endpoint path relative to the base actuator URL for the instance.
+	 * @return the contents as a {@code String}.
+	 */
+	default String getFromActuator(String deploymentId, int instanceIndex, String endpoint){
+		return getFromActuator(deploymentId, instanceIndex, endpoint, String.class);
+	}
+
+	/**
+	 * Get a resource from an actuator path, for single app instance with index 0.
+	 * @param deploymentId the deployment ID of the deployed app.
+	 * @param endpoint the endpoint path relative to the base actuator URL for the instance.
+	 * @return the contents as a {@code String}.
+	 */
+	default String getFromActuator(String deploymentId, String endpoint){
+		return getFromActuator(deploymentId, 0, endpoint);
+	}
+
+	/**
+	 * Post to resource on actuator path.
+	 * @param deploymentId the deployment ID of the deployed app.
+	 * @param instanceIndex index of the app instance.
+	 * @param endpoint the endpoint path relative to the base actuator URL for the instance.
+	 * @param body the request body.
+	 * @param responseType the expected response type.
+	 * @return the result (response body).
+	 */
+	<T,R> R postToActuator(String deploymentId, int instanceIndex, String endpoint, T body, Class<R> responseType);
+
+	/**
+	 * Post to resource on actuator path, for single app instance with index 0
+	 * @param deploymentId the deployment ID of the deployed app.
+	 * @param endpoint the endpoint path relative to the base actuator URL for the instance.
+	 * @param body the request body
+	 * @return the result (response body).
+	 */
+	default <T,R> R postToActuator(String deploymentId, String endpoint, T body, Class<R> responseType) {
+		return postToActuator(deploymentId, 0, endpoint, body, responseType);
+	}
+}
```

Comment:
Any need for [DELETE](https://docs.spring.io/spring-boot/docs/2.4.8/reference/htmlsingle/#production-ready-endpoints-custom-web-predicate-http-method)) endpoints or are we staying away from them because of the potential for destruction?

Reply:
It does map `@Delete` to `DELETE` and an example of a built-in delete endpoint is [here](https://docs.spring.io/spring-boot/docs/current/actuator-api/htmlsingle/#caches.evict-all).

Reply:
You're right. Good catch. I don't think we need it ATM.